homeassistant/components/vizio/media_player.py

```diff
@@ -410,6 +444,15 @@ def sound_mode_list(self) -> Optional[List[str]]:
         """List of available sound modes."""
         return self._available_sound_modes
 
+    @property
+    def device_state_attributes(self):
+        """Return additional state attributes."""
+        data = {ATTR_SETTING_TYPES: self._setting_types}
+        for setting_type, settings in self._all_settings.items():
```

Comment:
How big is this? Does it make sense to put this in the state machine or should this just be exposed as options flow?

Reply:
My speaker device has two setting types and overall around 15 settings, tv's will likely have more. Given that the settings won't change all that frequently, I would be OK moving this out of the state machine and into options, the tricky part would be building the schema for the options, which will require understanding the data type of each setting as well as the available options when the setting is a single select list. The advantage of sticking it in the state machine and exposing a service is that the integration does not need to know these things in order to allow users to update the settings.

Reply:
Here's what what be available for my device:
Audio settings:
```
INFO:pyvizio.cli:
Name              Value
----------------  -------
bass              0
treble            0
center            0
surround_volume   0
surround_balance  0
subwoofer         0
eq                Direct
surround_sound    On
height            On
height_level      4
volume_leveler    Off
night_mode        Off
av_delay          0
volume            5
mute              Off
```
System settings:
```
INFO:pyvizio.cli:
Name             Value
---------------  -------
menu_language    English
country          USA
vizio_tv_remote  Off
eco_power        Off
power_indicator  On
sleep_timer      Off
```

Reply:
This should not be pushed into an attribute but really just become (binary) sensors or switches if they can be changed? 

Or, if reading the values is not so important initially, you can just create a `vizio.set_config` service and that's it.

Reply:
Yeah that may the best plan (to create the service without creating the extra attributes). Will update the PR tomorrow